tsdb/index/tsi1/measurement_block.go

```diff
@@ -321,16 +326,17 @@ func (e *MeasurementBlockElem) TagBlockOffset() int64 { return e.tagBlock.offset
 // TagBlockSize returns the size of the measurement's tag block.
 func (e *MeasurementBlockElem) TagBlockSize() int64 { return e.tagBlock.size }
 
-// SeriesID returns series ID at an index.
-func (e *MeasurementBlockElem) SeriesID(i int) uint64 {
-	return binary.BigEndian.Uint64(e.series.data[i*SeriesIDSize:])
-}
-
 // SeriesIDs returns a list of decoded series ids.
 func (e *MeasurementBlockElem) SeriesIDs() []uint64 {
-	a := make([]uint64, e.series.n)
-	for i := 0; i < int(e.series.n); i++ {
-		a[i] = e.SeriesID(i)
+	a := make([]uint64, 0, e.series.n)
+	var prev uint64
+	for data := e.series.data; len(data) > 0; {
+		delta, n := binary.Uvarint(data)
```

Comment:
I was asking more about the range of values.  If it's < 128, varint encoding would use 2 bytes when 1 of a few bits are sufficient.  My guess is that series keys won't differ by a large range very frequently (since I believe they are sorted) so I think using a different encoding might save a lot of space here.

I think you mentioned offline that you were going to switch series ids back to `uint32`?  Is that still planned?